python/cavity_flow.py

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+from navier_stokes_problem import StationaryNavierStokesProblem, VelocityBCType
+from grid_generator import square_cavity, HyperCubeBoundaryMarkers
+
+class CavityProblem(StationaryNavierStokesProblem):
+    def __init__(self, n_points, main_dir = None):
+        super().__init__(main_dir)
+        
+        self._n_points = n_points
+        self._problem_name  = "Cavity"
+    
+    def setup_mesh(self):
+        # create mesh
+        self._mesh, self._boundary_markers = square_cavity(2, self._n_points)
+        
+    def set_boundary_conditions(self):
+        # velocity boundary conditions
+        velocity_bcs = (
+                (VelocityBCType.no_slip, HyperCubeBoundaryMarkers.left.value, None),
+                (VelocityBCType.no_slip, HyperCubeBoundaryMarkers.right.value, None),
+                (VelocityBCType.no_slip, HyperCubeBoundaryMarkers.bottom.value, None),
+                (VelocityBCType.constant, HyperCubeBoundaryMarkers.right.value, (1.0, 0.0)))
```

Comment:
for cavity_flow, like Lid-driven cavity, maybe the velocity boundary:
 (VelocityBCType.constant, HyperCubeBoundaryMarkers.right.value, (1.0, 0.0)))
  should be the top boundary not the right.

Reply:
Yes you are right. That's a stupid typo of mine. Fixed in efbb7f3.